src/Runner.Worker/ActionManager.cs

```diff
@@ -830,6 +831,12 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 if (!useActionArchiveCache)
                 {
                     await DownloadRepositoryArchive(executionContext, link, downloadInfo.Authentication?.Token, archiveFile);
+                    if (hasActionArchiveCache && externalCachingEnabled)
+                    {
+                        executionContext.Output($"Saving archive file to cache at '{cacheArchiveFile}'");
+                        Directory.CreateDirectory(Path.GetDirectoryName(cacheArchiveFile));
+                        File.Copy(archiveFile, cacheArchiveFile, true);
+                    }
```

Comment:
By adding the bit of code in the PR we are able to step into the middle of the actions download functionality and save the downloaded archive to the cache folder without any change to the rest of the logic flow. On subsequent runs, that archive is found and used by all runners as intended.
 
Further, we also attempted to mount the NFS as the temp folder, just as a “let’s try it and see what happens” kind of thing, knowing that we would be persisting far too much data. That also did not work.
 
According to our testing, a mounted NFS archive directory, regardless of “tool” or “action” cache HAS to exist in a directory outside of any of the directories created by the runner. This means that in order for a mounted persistent cache to work, the actions caching process HAS to function in the way that the tool cache functions e.g., persisting the downloaded archive to the provided cache directory.